packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -377,6 +377,28 @@ class ReactShallowRenderer {
       responder,
     });
 
+    // TODO: implement if we decide to keep the shallow renderer
+    const useTransition = (
+      config,
+    ): [(callback: () => void) => void, boolean] => {
+      this._validateCurrentlyRenderingComponent();
+      const startTransition = callback => {
+        callback();
+      };
+      warning(false, 'useTransition is not implemented for shallow renderers');
+      return [startTransition, false];
+    };
+
+    // TODO: implement if we decide to keep the shallow renderer
+    const useDeferredValue = <T>(value: T, config): T => {
+      this._validateCurrentlyRenderingComponent();
+      warning(
+        false,
+        'useDeferredValue is not implemented for shallow renderers',
+      );
```

Comment:
I don't think we need these warnings. We should be able to just remove them and keep the rest of this code as is.